### PR TITLE
Add Odin75 (Re-attempt)

### PR DESCRIPTION
**Pull request by adophoxia**
## Description

Re-entry of 19101 since the original PR maker has gone rogue and said PR has gone dormant for 6 months. Also implemented suggested changes from previous PR if applicable.
* #19101 

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [ ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/kbdfans/odin75/mcuconf.h [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+#pragma once
+
+#include_next <mcuconf.h>
+
+#undef RP_SPI_USE_SPI1
+#define RP_SPI_USE_SPI1 TRUE
+
+#undef RP_PWM_USE_PWM4
+#define RP_PWM_USE_PWM4 TRUE
```
Comment (zvecr) [inline keyboards/kbdfans/odin75/mcuconf.h:25]: Isnt used for anything?
```suggestion
```

File: keyboards/kbdfans/odin75/lib/bongocat.h [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
```
Comment (zvecr) [inline keyboards/kbdfans/odin75/lib/bongocat.h:16]:
```suggestion

#pragma once

```

File: keyboards/kbdfans/odin75/keymaps/via/keymap.c [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  ),
+
+  [3] = LAYOUT( /* Layer Zero */
+	KC_ESC,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_MUTE,  KC_PGUP, KC_PGDN,
+	KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC,  KC_HOME, KC_END,
+	KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS,  KC_INS,  KC_DEL,
+	KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,          KC_ENT,
+	KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT,          KC_PAUSE, KC_UP,
+	KC_LCTL, KC_LGUI, KC_LALT,                           KC_SPC,                                       KC_RALT, KC_RCTL, KC_LEFT,  KC_DOWN, KC_RGHT
+  )
```
Comment (zvecr) [inline keyboards/kbdfans/odin75/keymaps/via/keymap.c:54]:
```suggestion
  )
```

File: keyboards/kbdfans/odin75/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+  ),
+
+  [3] = LAYOUT( /* Layer Zero */
+	KC_ESC,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_MUTE,  KC_PGUP, KC_PGDN,
+	KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC,  KC_HOME, KC_END,
+	KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS,  KC_INS,  KC_DEL,
+	KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,          KC_ENT,
+	KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT,          KC_PAUSE, KC_UP,
+	KC_LCTL, KC_LGUI, KC_LALT,                           KC_SPC,                                       KC_RALT, KC_RCTL, KC_LEFT,  KC_DOWN, KC_RGHT
+  )
```
Comment (zvecr) [inline keyboards/kbdfans/odin75/keymaps/default/keymap.c:54]:
```suggestion
  )
```

File: keyboards/kbdfans/odin75/halconf.h [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+#define HAL_USE_SPI TRUE
+
+#undef SPI_USE_WAIT
+#define SPI_USE_WAIT TRUE
+
+#undef SPI_SELECT_MODE
+#define SPI_SELECT_MODE SPI_SELECT_MODE_PAD
+
+#undef HAL_USE_PWM
+#define HAL_USE_PWM TRUE
```
Comment (zvecr) [inline keyboards/kbdfans/odin75/halconf.h:31]:
```suggestion
```

File: keyboards/kbdfans/odin75/keymaps/default/keymap.c [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+  [0] = LAYOUT( /* Layer Zero */
```
Comment (zvecr) [inline keyboards/kbdfans/odin75/keymaps/default/keymap.c:20]:
```suggestion
  [0] = LAYOUT(
```

File: keyboards/kbdfans/odin75/keymaps/via/keymap.c [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+  [0] = LAYOUT( /* Layer Zero */
```
Comment (zvecr) [inline keyboards/kbdfans/odin75/keymaps/via/keymap.c:20]:
```suggestion
  [0] = LAYOUT(
```

File: keyboards/kbdfans/odin75/odin75.c [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+
+#include "quantum.h"
+#include "lib/bongocat.h"
+#ifdef OLED_ENABLE
+  oled_rotation_t oled_init_user(oled_rotation_t rotation) {
+  //return OLED_ROTATION_180;
+  return OLED_ROTATION_0;
+}
+    bool oled_task_kb(void) {
+        if (!oled_task_user()) { return false; }
```
Comment (zvecr) [inline keyboards/kbdfans/odin75/odin75.c:25]: Indentation should be fixed

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/kbdfans/odin75/odin75.c [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+        oled_set_cursor(14, 0); // sets cursor to (column, row) using charactar spacing (4 rows on 128x32 screen, anything more will overflow back to the top)
+        oled_write_P(PSTR("WPM:"), false);
+        oled_write(get_u8_str(get_current_wpm(), '0'), false); // writes wpm on top right corner of string
+        oled_set_cursor(0, 5);
+        oled_write_P(led_state.caps_lock ? PSTR("CAPS") : PSTR("    "), false);
+        oled_set_cursor(0, 6);
+        oled_write_P(led_state.num_lock ? PSTR("NUM") : PSTR("    "), false);
+        oled_set_cursor(0, 7);
+        oled_write_P(led_state.scroll_lock ? PSTR("SCRL") : PSTR("    "), false);
+    }
```
Comment (zvecr) [inline keyboards/kbdfans/odin75/odin75.c:37]:
```suggestion
    return false;
}
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/kbdfans/odin75/odin75.c (hunk 8 above)
Comment (adophoxia) [reply keyboards/kbdfans/odin75/odin75.c]: Fixed in 9947ebb0d12c579ec771795a3e5372f8ac68c9ab. (Used tabs for spacing)

File: keyboards/kbdfans/odin75/odin75.c (hunk 9 above)
Comment (adophoxia) [reply keyboards/kbdfans/odin75/odin75.c]: Resolved in 9865af8fca062e7b6f3a15c77a0997620ff8148b.

File: keyboards/kbdfans/odin75/odin75.c [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+* along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "quantum.h"
+#include "lib/bongocat.h"
+#ifdef OLED_ENABLE
+	oled_rotation_t oled_init_user(oled_rotation_t rotation) {
+	//return OLED_ROTATION_180;
+	return OLED_ROTATION_0;
+}
```
Comment (waffle87) [inline keyboards/kbdfans/odin75/odin75.c:23]: Default function behaviour.
```suggestion
```

File: keyboards/kbdfans/odin75/keymaps/via/keymap.c [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+  [0] = LAYOUT(
+	KC_ESC,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_MUTE,  KC_PGUP, KC_PGDN,
+	KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC,  KC_HOME, KC_END,
+	KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS,  KC_INS,  KC_DEL,
+	KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,          KC_ENT,
+	KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT,          KC_PAUSE, KC_UP,
+	KC_LCTL, KC_LGUI, KC_LALT,                           KC_SPC,                                       KC_RALT, KC_RCTL, KC_LEFT,  KC_DOWN, KC_RGHT
```
Comment (drashna) [inline keyboards/kbdfans/odin75/keymaps/via/keymap.c:26]:
```suggestion
    KC_ESC,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_MUTE,  KC_PGUP, KC_PGDN,
    KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC,  KC_HOME, KC_END,
    KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS,  KC_INS,  KC_DEL,
    KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,          KC_ENT,
    KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT,          KC_PAUSE, KC_UP,
    KC_LCTL, KC_LGUI, KC_LALT,                           KC_SPC,                                       KC_RALT, KC_RCTL, KC_LEFT,  KC_DOWN, KC_RGHT
```

File: keyboards/kbdfans/odin75/keymaps/default/keymap.c [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+  [0] = LAYOUT(
+	KC_ESC,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_MUTE,  KC_PGUP, KC_PGDN,
+	KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC,  KC_HOME, KC_END,
+	KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS,  KC_INS,  KC_DEL,
+	KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,          KC_ENT,
+	KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT,          KC_PAUSE, KC_UP,
+	KC_LCTL, KC_LGUI, KC_LALT,                           KC_SPC,                                       KC_RALT, KC_RCTL, KC_LEFT,  KC_DOWN, KC_RGHT
```
Comment (drashna) [inline keyboards/kbdfans/odin75/keymaps/default/keymap.c:26]:
```suggestion
    KC_ESC,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_MUTE,  KC_PGUP, KC_PGDN,
    KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC,  KC_HOME, KC_END,
    KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS,  KC_INS,  KC_DEL,
    KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,          KC_ENT,
    KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT,          KC_PAUSE, KC_UP,
    KC_LCTL, KC_LGUI, KC_LALT,                           KC_SPC,                                       KC_RALT, KC_RCTL, KC_LEFT,  KC_DOWN, KC_RGHT
```

File: keyboards/kbdfans/odin75/odin75.c [hunk 13]
```diff
@@ -0,0 +1,35 @@
...
+	oled_set_cursor(14, 0); // sets cursor to (column, row) using charactar spacing (4 rows on 128x32 screen, anything more will overflow back to the top)
+	oled_write_P(PSTR("WPM:"), false);
+	oled_write(get_u8_str(get_current_wpm(), '0'), false); // writes wpm on top right corner of string
+	oled_set_cursor(0, 5);
+	oled_write_P(led_state.caps_lock ? PSTR("CAPS") : PSTR("    "), false);
+	oled_set_cursor(0, 6);
+	oled_write_P(led_state.num_lock ? PSTR("NUM") : PSTR("    "), false);
+	oled_set_cursor(0, 7);
+	oled_write_P(led_state.scroll_lock ? PSTR("SCRL") : PSTR("    "), false);
+	return false;
```
Comment (drashna) [inline keyboards/kbdfans/odin75/odin75.c:33]:
```suggestion
    if (!oled_task_user()) { return false; }
    led_t led_state = host_keyboard_led_state();
    render_bongocat();
    oled_set_cursor(14, 0); // sets cursor to (column, row) using charactar spacing (4 rows on 128x32 screen, anything more will overflow back to the top)
    oled_write_P(PSTR("WPM:"), false);
    oled_write(get_u8_str(get_current_wpm(), '0'), false); // writes wpm on top right corner of string
    oled_set_cursor(0, 5);
    oled_write_P(led_state.caps_lock ? PSTR("CAPS") : PSTR("    "), false);
    oled_set_cursor(0, 6);
    oled_write_P(led_state.num_lock ? PSTR("NUM") : PSTR("    "), false);
    oled_set_cursor(0, 7);
    oled_write_P(led_state.scroll_lock ? PSTR("SCRL") : PSTR("    "), false);
    return false;
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by drashna** [comment]
@adophoxia just a heads up, you can add multiple suggestions to a batch commit on the files changed tab.  Wasn't sure if you were aware of that or not.

**Review by drashna — APPROVED** [review]

**Comment by adophoxia** [comment]
@drashna, tried to do so in the past for other suggestions, but GitHub was being stupid with me and wouldn't let me do so. Thx for the heads-up tho.

File: keyboards/kbdfans/odin75/readme.md [hunk 14]
```diff
@@ -0,0 +1,22 @@
+# ODIN75
+A customizable 75% keyboard.
+
+* Keyboard Maintainer: [moyi4681](https://github.com/moyi4681)
+* Hardware Supported: ODIN75
+* Hardware Availability: [KBDfans](https://kbdfans.com/)
```
Comment (waffle87) [inline keyboards/kbdfans/odin75/readme.md:6]:
```suggestion
* Hardware Availability: [KBDfans](https://kbdfans.com/products/kbdfans-odin-75-mechanical-keyboard)
```

File: keyboards/kbdfans/odin75/readme.md [hunk 15]
```diff
@@ -0,0 +1,22 @@
...
+
+* Bootmagic reset: Hold down the left top key then plug in the USB cable.
+* Physical reset button: Hold down the reset button then plug in the USB cable.
+* Keycode in layout: Press the key mapped to QK_BOOT if it is available.
+
+## Making firmware
+
+Make example for this keyboard (after setting up your build environment):
+
+    make kbdfans/odin75:default
```
Comment (waffle87) [inline keyboards/kbdfans/odin75/readme.md:20]:
```suggestion
    make kbdfans/odin75:default

Flashing example for this keyboard:

    make kbdfans/odin75:default:flash
```

**Review by waffle87 — APPROVED** [review]

**Pull request merged**